### ISSUE TITLE
Convert RpcRunner to definition-based model

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#119]
Originally reported of Google Code with ID 63.

```
Currently it's very hard to handle errors in RpcRunner functions in a
general way, especially for serializing or deserializing. Converting it to
a definition-based model should help with this.

Just some ideas in pseudocode:

class RpcRunner(…):
  def _SerializeDisks(disks):
    return [disk.ToDict() for disk in disks]

  def _DeserializeBlockdevStatus(status):
    return [objects.BlockDevStatus.FromDict(i)
            for i in status]

  (MULTI_NODE, SINGLE_NODE) = range(2)

  RPC_FUNCTIONS = {
    # name: (type, [arg1_serializer, arg2_serializer],
    #        result_deserializer)
    # serializer and deserializer can be None
    "blockdev_getmirrorstatus": (SINGLE_NODE,
      [_SerializeDisks], _DeserializeBlockdevStatus),
    }
```
Originally added on 2009-08-05 11:40:57 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2009-08-05 12:12:17 +0000 UTC.

Added Labels: Priority-Medium Component-master

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2010-03-12 19:39:33 +0000 UTC.

Added to Milestone: Unplanned

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2011-08-05 14:45:42 +0000 UTC.

Removed from Milestone: Unplanned

**Comment by Ganeti-Issues-Migrator** [comment]
```
This has been implemented in the 2.6 release.
```
Originally added on 2012-07-19 12:55:17 +0000 UTC.

Added Labels: Type-Refactoring
Removed Labels: Type-Enhancement
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-22 08:57:07 +0000 UTC.

Changed State: Released
